### DISCUSSION TITLE
Error messages trying to compute ensemble statistics

### DISCUSSION THREAD

**Discussion by knlowman** [Ouranosinc/xclim#1923] [Questions] [answered]
### Setup Information

* Xclim version: 0.52.1

### Context

I am trying to compute the ensemble statistics for the dataset `ctrl_ens`, which looks like this:
![Screenshot 2024-09-16 at 10 42 57 AM](https://github.com/user-attachments/assets/46f54c73-89f7-4355-acc6-b79fa4be4b52)

### Steps To Reproduce

My code for computing stats for the entire dataset is
```
ctrl_ens = ensembles.create_ensemble([ctrl_mem_1,ctrl_mem_2,ctrl_mem_3])
ensembles.ensemble_mean_std_max_min(ctrl_ens)
```
The corresponding error message is
`UFuncTypeError: ufunc 'add' cannot use operands with types dtype('<M8[ns]') and dtype('<M8[ns]')`.

I also tried computing stats just for the variable `temp` with `ensembles.ensemble_mean_std_max_min(ctrl_ens.temp)`
and I got the error `AttributeError: 'DataArray' object has no attribute 'data_vars'`.

**Comment by aulemahal** [comment, marked as answer]
Hi @knlowman,

For both errors, I think one issue is that you didn't expect that `ensemble_mean_std_max_min` takes ONLY Datasets as input. We don't have a real reason for this, the function could easily be made to accept `DataArray` as well (a good first contribution maybe ;) )?

For the second error, try doing :
```python3
ensembles.ensemble_mean_std_max_min(ctrl_ens[['temp']])
```
Here I am using xarray's feature of selecting a list of (data) variables, which returns a Dataset with only that (data) variable. Keeping all relevant coordinates and attributes.

And the first error. `dtype('<M8[ns]')` means that one of the data variables in your dataset is made of datetimes. Are you expecting of taking the mean/std/max/min of dates ? I am guessing you are not, so the solution would be to exclude these variables from the computation. Using the same trick as above, you could filter out only the meaningful variables.

If you are indeed expecting stats on dates, then the problematic stat is `std`. You could make the calculation outside the xclim function for the particular datetime variables, skipping `std` as it is ill-defined for datetime.

**Reply by knlowman** [reply]
Thanks very much @aulemahal! That solved the second error, and you're correct that I don't want to average a datetime variable, so I excluded it.
